freqtrade/freqtradebot.py

```diff
@@ -1596,9 +1598,19 @@ def update_trade_state(self, trade: Trade, order_id: str, action_order: Dict[str
         Trade.commit()
 
         if order['status'] in constants.NON_OPEN_EXCHANGE_STATES:
-            # If a buy order was closed, force update on stoploss on exchange
-            if order.get('side', None) == 'buy':
+            # If a entry order was closed, force update on stoploss on exchange
+            if order.get('side', None) == trade.enter_side:
                 trade = self.cancel_stoploss_on_exchange(trade)
+                # TODO: Margin will need to use interest_rate as well.
+                _, isolated_liq = self.leverage_prep(
+                    leverage=trade.leverage,
+                    pair=trade.pair,
+                    amount=trade.amount,
+                    open_rate=trade.open_rate,
+                    is_short=trade.is_short
+                )
+                if isolated_liq:
```

Comment:
@samgermain calling leverage_prep here is the right place (it's when the order filled - so we now know the "real" amount).

The question is though - would this be sufficient overall? (so we could remove the call while placing the order? (where the amount may still change, in case of timedout orders) ?

Reply:
> 

I think the point of the method was to avoid flake8 "This method is too complicated" errors. The error might be gone if I comment out the leverage parts though

---

Is that what you're asking?

Reply:
well - no.
I'm asking you (and myself as well) if we really need to call this method from 2 places in freqtradebot.

I think calling it here (when the order is effectively filled) is sufficient.
It's not relevant while placing the order - only while handling the trade.

Reply:
I'll add a todo there to doublecheck - it's not something we need to check/decide now i think.